@navikt/core/react/src/overlays/dismissablelayer/DismissableLayer.tsx

```diff
@@ -54,156 +57,116 @@ interface DismissableLayerBaseProps {
-     * To correctly handle nested DismissableLayer,
-     * we only initialize the `Descendants`-API for the root layer to aboid resetting context
-     */
-    return context ? (
-      <DismissableLayerNode ref={ref} {...props} />
-    ) : (
-      <DismissableRoot>
-        <DismissableLayerNode ref={ref} {...props} />
-      </DismissableRoot>
-    );
+    return <DismissableLayerInternal {...restProps} ref={forwardedRef} />;
   },
 );
 
-/**
- * DismissableRoot
- *
- * Used to initialize the `Descendants`-API at the root layer.
- * All subsequent layers will use the same context.
- */
-const DismissableRoot = ({ children }: { children: React.ReactNode }) => {
-  const descendants = useDismissableDescendants();
+type DismissableLayerElement = React.ComponentRef<
+  typeof DismissableLayerInternal
+>;
+
+const BranchedLayerContext =
+  React.createContext<DismissableLayerElement | null>(null);
+
+/* ------------------------ DismissableLayerInternal ------------------------ */
+const CONTEXT_UPDATE_EVENT = "dismissableLayer.update";
+let originalBodyPointerEvents: string;
 
-  return (
-    <DismissableDescendantsProvider value={descendants}>
-      {children}
-    </DismissableDescendantsProvider>
-  );
-};
+const DismissableLayerContext = React.createContext({
+  layers: new Set<DismissableLayerElement>(),
+  branchedLayers: new Map<
+    DismissableLayerElement,
+    Set<DismissableLayerElement>
+  >(),
+  layersWithOutsidePointerEventsDisabled: new Set<DismissableLayerElement>(),
+});
 
-const DismissableLayerNode = forwardRef<HTMLDivElement, DismissableLayerProps>(
+const DismissableLayerInternal = forwardRef<
+  HTMLDivElement,
+  DismissableLayerProps
+>(
   (
     {
       children,
-      asChild,
+      disableOutsidePointerEvents,
+      onDismiss,
+      onInteractOutside,
       onEscapeKeyDown,
-      onPointerDownOutside,
       onFocusOutside,
-      onInteractOutside,
-      onDismiss,
+      onPointerDownOutside,
       safeZone,
-      disableOutsidePointerEvents = false,
-      enabled = true,
-      ...rest
+      asChild,
+      ...restProps
     }: DismissableLayerProps,
-    ref,
+    forwardedRef,
   ) => {
-    const [, setForce] = useState({});
-    const { register, index, descendants } = useDismissableDescendant({
-      disableOutsidePointerEvents,
-      disabled: !enabled,
-      forceUpdate: () => setForce({}),
-    });
-
-    /**
-     * `node` will be set to the ref of the component or nested component
-     * Ex: If
-     * ```
-     * <DismissableLayer asChild>
-     *   <Popover />
-     * </DismissableLayer>
-     * ```
-     * `node` will in this case be the Popover-element.
-     * We use State her and not ref since we want to trigger a rerender when the node changes.
-     */
-    const [node, setNode] = useState<HTMLDivElement | null>(null);
-
-    const mergedRefs = useMergeRefs(setNode, register, ref);
+    const context = useContext(DismissableLayerContext);
 
+    const [, force] = useState({});
```

Comment:
The setter could have a clearer name

```suggestion
    const [, forceRerender] = useState({});
```